src/logger.ts

```diff
@@ -0,0 +1,93 @@
+import * as os from 'os';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as winston from 'winston';
+import * as TransportStream from 'winston-transport';
+
+class OutputChannelTransport extends TransportStream {
+    outputChannel: vscode.OutputChannel;
+
+    constructor(outputChannel: vscode.OutputChannel) {
+        super();
+        this.outputChannel = outputChannel;
+    }
+
+    log(info, callback) {
+        setImmediate(() => {
+            this.emit('logged', info);
+        });
+
+        this.outputChannel.appendLine(`[${info.label}] ${info.level}: ${info.message}`);
+
+        if (callback) {
+            callback();
+        }
+    }
+}
+
+const fileLoggerFormat = winston.format.printf(({ level, message, label, timestamp }) => {
+    return `${timestamp} [${label}] ${level}: ${message}`;
+});
+
+let fileLogger;
+let consoleLogger;
+let isInitialised: Boolean = false;
+
+export function initLogging(outputChannel: vscode.OutputChannel, filename?: string) {
+    const format = winston.format.combine(
+        winston.format.label({ label: 'nisaba' }),
+        winston.format.timestamp(),
+        fileLoggerFormat
+    );
+
+    if (filename === undefined) {
+        // When no `filename` is passed in, default to `~/.nisaba/nisaba.log`
+        filename = path.join(os.homedir(), '.nisaba', 'nisaba.log')
+    }
+
+    fileLogger = winston.createLogger({
+        level: 'debug',
+        format: format,
+        defaultMeta: { service: 'user-service' },
+        transports: [
+            new winston.transports.File({ filename: filename }),
+        ],
+    });
+
+    consoleLogger = winston.createLogger({
+        level: 'info',
+        format: format,
+        defaultMeta: { service: 'user-service' },
+        transports: [
+            new OutputChannelTransport(outputChannel),
+        ],
+    });
+
+    isInitialised = true;
```

Comment:
Should the whole body be skipped if `isInitialised` is already true? Do we want to warn (for our benefit)? In case we do more intricate logging in the future...

Reply:
Done!